docs/latest/design/extending-envoy-gateway.md

```diff
@@ -0,0 +1,147 @@
+    service:
+      host: dummy-extension.example
+      port: 443
+      tls:
+        type: secret
+        secret:
+            name: dummy-extension
+            namespace: example
+```
+
+## Data Plane Extensions
+
+Envoy Gateway manages [Envoy][] deployments, which acts as the data plane that handles actual user traffic. Users configure the data plane using the K8s Gateway API resources which Envoy 
+Gateway converts into [Envoy specific configuration (xDS)][] to send over to Envoy.
+
+Gateway API offers [extensionRef filters][] and [Policy Attachments][] as extension points for implementers to use. Envoy Gateway extends the Gateway API using these extension points to provide support for [rate limiting][] 
+and [authentication][]. The initial design of Envoy Gateway extensions will primarily focus on extensionRef filters, where users can reference a custom resource as an HTTP Filter.
+
+This allows users to be able to reference additional filters provided by extensions to Envoy Gateway, in their [HTTPRoutes][]:
+
+```yaml
+apiVersion: example.myextension.io/v1alpha1
+kind: OAuth2Filter
+metadata:
+  name: oauth2-filter
+spec:
+  ...
+
+---
+apiVersion: gateway.networking.k8s.io/v1beta1
+kind: HTTPRoute
+metadata:
+  name: example
+spec:
+  parentRefs:
+  - name: eg
+  hostnames:
+  - www.example.com
+  rules:
+  - clientSelectors:
+    - path:
+        type: PathPrefix
+        value: /
+    filters:
+    - type: ExtensionRef
+      extensionRef:
+        group: example.myextension.io
+        kind: OAuth2Filter
+        name: oauth2-filter
+    backendRefs:
+    - name: backend
+      port: 3000
+```
+
+In order to enable this, Envoy Gateway provides hook entrypoints within the translation pipeline, where it calls out to extension services registered in the [EnvoyGateway config][] 
+if they specify an `apiGroup` that matches the `group` of an `ExtensionRef` filter. The extension will then be able to modify the xDS that Envoy Gateway generated and send back the 
+modified configuration. If there are no registered extensions that specify support for the `group` of an `ExtensionRef` filter then Envoy Gateway will treat it as an unknown resource 
+and provide an error to the user.
+
+## Control Plane Extensions
+
+To provide certain extensions such as a new authentication filter it is also necessary to be able to extend the Control plane
+to deploy additional services that are required such as a new [external authorization][] server or rate limiting server.
+
+To power these use-cases, Envoy Gateway provides additional hook entrypoints to the Infra Manager which is responsible for dynamically creating the
+instances of Envoy Proxy. These hook points empower extensions to modify the Envoy Proxy deployment and service dynamically.
+
+TODO: I think it's fine if we keep this section but have we considered whether the use-case of extending the Infra would be better served by an addition to the EnvoyProxy API to allow
```

Comment:
I think this is a good point

Reply:
I don't have any strong opinions...let's maybe see what additional feedback we get.